dask/array/core.py

```diff
@@ -2747,7 +2747,12 @@ def squeeze(self, axis=None):
         return squeeze(self, axis)
 
     def rechunk(
-        self, chunks="auto", threshold=None, block_size_limit=None, balance=False
+        self,
+        chunks="auto",
+        threshold=None,
+        block_size_limit=None,
+        balance=False,
+        rechunk=None,
```

Comment:
I don't love the name for this. Naming is hard. I see three options

1. rechunk (as you are proposing)
2. algo (more generic but also more descriptive)
3. shuffle (not great but it would offer some consistency; probably the worst option :D )

Reply:
I've chosen #1 for consistency with the `shuffle` keyword, which I'm not a huge fan of either.

The most descriptive might be 
4. `rechunk_{impl|algo}`

This would be more descriptive and generalizes well to other methods that use a `rechunk` under the hood, similar to how we use `shuffle` in other functions as well. `impl` might give us slightly more freedom (I could see different implementations of a P2P-like algorithm, e.g., an RDMA-based one).

I'd even go as far as proposing to rename the `shuffle` kw.

Reply:
> In that case, I'd even go as far as proposing to rename the shuffle keyword.

That's also on the table (with a deprecation cycle, of course)

Reply:
It's now `algorithm` as a keyword within `rechunk`, which should be named `rechunk_algorithm` within users of `rechunk`. In the config, it's `array.rechunk.algorithm`.